File: banda/JCZhang-Banda/JCZhang-Banda-JAVA/src/main/java/models/SingerGuitarist.java
package models;

import interfaces.IGuitarist;
import interfaces.ISinger;

public class SingerGuitarist extends Musician implements ISinger, IGuitarist {

    String name;
    int yearsOfExperience;
    String tone;
    String guitarType;

    static double salary = (Musician.salary * 1.5);

    public SingerGuitarist(String name, int yearsOfExperience, String tone, String guitarType) {
        super(name, yearsOfExperience);
        this.name = name;
        this.yearsOfExperience = yearsOfExperience;
        this.tone = tone;
        this.guitarType = guitarType;

    }

    @Override
    public void playGuitar() {
        System.out.println("Im a SingerGuitarist and im playing guitar");
    }

    @Override
    public void sing() {
        System.out.println("Im a SingerGuitarist and im singing");
    }
}

File: banda/JCZhang-Banda/JCZhang-Banda_KOTLIN/src/main/kotlin/models/SingerGuitarist.kt
package models

import interfaces.IGuitarist
import interfaces.ISinger

class SingerGuitarist(
    override var name: String?,
    override var yearsOfExperience: Int,
    var tone: String,
    var guitarType: String
) :
    Musician(name, yearsOfExperience), ISinger, IGuitarist {
    override fun playGuitar() {
        println("Im a SingerGuitarist and im playing guitar")
    }

    override fun sing() {
        println("Im a SingerGuitarist and im singing")
    }

    companion object {
        var salary = Musician.salary * 1.5
    }
}